### PR TITLE
fix(core): set taskrun value in workingDirectory children task

### PR DESCRIPTION
**Pull request by Skraye**
Recap of what has been done and why : 

* Working Directory having a "custom task path," the working Directory task itself wasn't added in the `parents` and `parent` variables, preventing access to the `parent.taskrun.value` that should exist when in a flowable

* My mistake was to use a two nested flowable test for the debugging, so this time, I make a more straightforward test that uses `taskrun.value` 

* I then realized that the taskrun property of the children task was overridden in the runAttempts, and as a workingDirectory can not have flowables inside (meaning children of a `workingDirectory` could never have a `value` inside their `taskrun` property), 
* I created a new method on top of the `forWorker`  to keep logic separated, that will check if the runContext contains the initialized `workerTaskRun` exists and provide its `value` if has one.

TLDR : 
Every task in a `workingDirectory` now have their `taskrun.value` property set if` workingDirectory` is in a flowable that provides a `taskrun.value` so its transparent either you're using a `workingDirectory` or no

### CODE REVIEW

**Comment by Skraye** [comment]
close #1809

File: core/src/main/java/io/kestra/core/runners/RunContext.java [hunk 1]
```diff
@@ -440,18 +440,31 @@ public RunContext forWorkerDirectory(ApplicationContext applicationContext, Work
...
-        if (clone.containsKey("parent")) {
-            clone.put("parent", clone.get("parent"));
-        }
 
         this.variables = ImmutableMap.copyOf(clone);
 
         return this;
     }
 
+    public RunContext forWorkerHandleDirectoryTask(ApplicationContext applicationContext, WorkerTask workerTask) {
```
Comment (loicmathieu) [inline core/src/main/java/io/kestra/core/runners/RunContext.java:449]: Why did you create a new method? Why don't you update the existing one as it is only used in worker so safe to update

**Review by loicmathieu — CHANGES_REQUESTED** [review]

**Review by loicmathieu — APPROVED** [review]
LGTM
